app/controllers/concerns/find_common.rb

```diff
@@ -11,7 +11,7 @@ def find_resource
   def resource_finder(scope, id)
     raise ActiveRecord::RecordNotFound if scope.empty?
     result = scope.from_param(id) if scope.respond_to?(:from_param)
-    result ||= scope.friendly.find(id) if scope.respond_to?(:friendly)
+    result ||= scope.friendly.find(id) rescue nil if scope.respond_to?(:friendly)
```

Comment:
I actually found this issue because of this test: https://github.com/theforeman/foreman/blob/develop/test/functional/api/v2/models_controller_test.rb#L46, do you think we need another one?
I'm not sure if another exception might be raised, but I'm also not sure we care why friendly find failed-  we just fallback to find by id, and if the result is still empty after everything then a 404 will be returned anyways.